docs: add "first steps" page and reduce h3 spacing (#703)
t3-oss/create-t3-app: www/src/pages/en/usage/first-steps.md

---
title: First Steps
description: Getting started with your new T3 App
layout: ../../../layouts/docs.astro
---

You just scaffolded a new T3 App and are ready to go. Here is the bare minimum to get your app working.

## Database

> If your app includes Prisma, make sure to run `npx prisma db push` from the root directory of your app. Note that you need to restart the dev server after doing this.

Reviewer: ```suggestion
If your app includes Prisma, make sure to run `npx prisma db push` from the root directory of your app. This command will sync your Prisma schema with your database and will generate the TypeScript types for the Prisma client based on your schema. Note that you need to restart the TypeScript server after doing this so that it can detect the generated types.
```

## Authentication

> If your app includes NextAuth, we get you started with the `DiscordProvider`. This is one of the simplest providers that NextAuth offers, but it still requires a bit of initial setup on your part.

Reviewer 2: ```suggestion
If your app includes NextAuth, we get you started with the `DiscordProvider`. This is one of the simplest providers that NextAuth.js offers, but it still requires a bit of initial setup on your part.
```

Reviewer 2: Looks like NextAuth.js is the right one. https://github.com/t3-oss/create-t3-app/pull/707#issuecomment-1307218276

> If your app includes Next-Auth, we get you started with the `DiscordProvider`. This is one of the simplest providers that Next-Auth offers, but it still requires a bit of initial setup on your part.

Reviewer 2: ```suggestion
If your app includes NextAuth, we get you started with the `DiscordProvider`. This is one of the simplest providers that NextAuth offers, but it still requires a bit of initial setup on your part.
```

Reviewer 2: The [NextAuth.js](https://next-auth.js.org/getting-started/example#install-nextauth) documentation uses `NextAuth` instead of `Next-Auth`. We should use the same IMO.
![NextAuth](https://user-images.githubusercontent.com/89210438/200529645-ff5c5833-fb65-4398-afa0-181dc16c425a.png)

Reviewer 2: Looks like the entire docs uses `Next-Auth`, and `NextAuth` interchangeably. I will make a PR later to fix all of them for consistency if that's alright.

> Of course, if you prefer to use a different auth provider, you can also use one of the [many providers](https://next-auth.js.org/providers/) that NextAuth offers.

Reviewer 2: ```suggestion
Of course, if you prefer to use a different auth provider, you can also use one of the [many providers](https://next-auth.js.org/providers/) that NextAuth.js offers.
```

> Of course if you prefer to use a different auth provider, you can also use one of the [many providers](https://next-auth.js.org/providers/) that Next-Auth offers.

Reviewer 2: ```suggestion
Of course, if you prefer to use a different auth provider, you can also use one of the [many providers](https://next-auth.js.org/providers/) that NextAuth offers.
```

1. You will need a Discord account, so register one if you haven't already.
2. Navigate to https://discord.com/developers/applications and click "New Application" in the top right corner. Give your application a name and agree to the Terms of Service.
3. Once your application has been created, navigate to "Settings → OAuth2 → General".
4. Copy the "Client ID" and add it to your `.env` as `DISCORD_CLIENT_ID`.
5. Click "Reset Secret", copy the new secret, and add it to your `.env` as `DISCORD_CLIENT_SECRET`.

> 6. Click "Add Redirect" and type in `http://localhost:3000/api/auth/callback/discord`.

Reviewer: ```suggestion
6. Click "Add Redirect" and type in `http://localhost:3000/api/auth/callback/discord`. You will also need to add the production URL here when you deploy the app.
```

Author: IMO it's better to create an entirely different app for dev/prod, will write something that mentions both options maybe?

Reviewer: Okay then it makes sense I guess

> 7. Save Changes. You should now be able to log in. Note that you need to restart the dev server for changes in environment variables to be reflected in your app.

Reviewer 3: IIRC, .env's are also hot reloaded nowadays?

Author: just tried it and you're right. didn't know that, amazing!

Author: changed in https://github.com/t3-oss/create-t3-app/pull/703/commits/58cbcb68c0e7eb38fc2d9c51053ccf19477a9d2a

## Next Steps

* If your app includes tRPC, check out `src/pages/index.tsx` and `src/server/trpc/router/example.ts` to see how tRPC queries work.
* Have a look around the `create-t3-app` docs, as well as the docs of the packages that your app includes.

> * Join the [Discord](https://t3.gg/discord) and give us a star on [GitHub](https://github.com/t3-oss/create-t3-app) :^)

Reviewer: ```suggestion
Join the [Discord](https://t3.gg/discord) and give us a star on [GitHub](https://github.com/t3-oss/create-t3-app) :)
````

not the nose 😭 lol

Author: sorry
<img width="813" alt="image" src="https://user-images.githubusercontent.com/8353666/200496940-99a950cd-d7ee-4f96-893d-d958397c9653.png">

---
title: First Steps
description: Getting started with your new T3 App
layout: ../../../layouts/docs.astro
---

You just scaffolded a new T3 App and are ready to go. Here is the bare minimum to get your app working.

## Database

If your app includes Prisma, make sure to run `npx prisma db push` from the root directory of your app. This command will sync your Prisma schema with your database and will generate the TypeScript types for the Prisma Client based on your schema. Note that you need to restart the TypeScript server after doing this so that it can detect the generated types.

## Authentication

> If your app includes NextAuth, we get you started with the `DiscordProvider`. This is one of the simplest providers that NextAuth.js offers, but it still requires a bit of initial setup on your part.

Reviewer 2: this one too

1. You will need a Discord account, so register one if you haven't already.
2. Navigate to https://discord.com/developers/applications and click "New Application" in the top right corner. Give your application a name and agree to the Terms of Service.
3. Once your application has been created, navigate to "Settings → OAuth2 → General".
4. Copy the "Client ID" and add it to your `.env` as `DISCORD_CLIENT_ID`.
5. Click "Reset Secret", copy the new secret, and add it to your `.env` as `DISCORD_CLIENT_SECRET`.
6. Click "Add Redirect" and type in `http://localhost:3000/api/auth/callback/discord`.
   - For production deployment, follow the previous steps to create another Discord Application, but this time replace `http://localhost:3000` with the URL that you are deploying to.
7. Save Changes. You should now be able to log in.

## Next Steps

- If your app includes tRPC, check out `src/pages/index.tsx` and `src/server/trpc/router/example.ts` to see how tRPC queries work.
- Have a look around the `create-t3-app` docs, as well as the docs of the packages that your app includes.

> - Join the [Discord](https://t3.gg/discord) and give us a star on [GitHub](https://github.com/t3-oss/create-t3-app)! :)

Reviewer 2: ```suggestion
- Join our [Discord](https://t3.gg/discord) and give us a star on [GitHub](https://github.com/t3-oss/create-t3-app)! :)
```

Reviewer 2: nit
